src/index.ts

```diff
@@ -1,32 +1,67 @@
-/** Author: yusufhilmi
+/** Author: yusufhilmi, lurrobert
  * Let's keep everything here until it gets too big
  * My first npm package, can use help about design patterns, best practices!
  */
 
+import Cache from './cache';
+import { IndexedDbManager } from './indexedDB';
+
+// uncomment for testing only
+// import { IDBFactory } from 'fake-indexeddb';
+// const indexedDB = new IDBFactory();
+
+interface SearchResult {
+  similarity: number;
+  object: any;
+}
+
+type StorageOptions = 'indexedDB' | 'localStorage' | 'none';
+
+/**
+ * Interface for search options in the EmbeddingIndex class.
+ * topK: The number of top similar items to return.
+ * filter: An optional filter to apply to the objects before searching.
+ * useStorage: A flag to indicate whether to use storage options like indexedDB or localStorage.
+ */
+interface SearchOptions {
+  topK?: number;
+  filter?: { [key: string]: any };
+  useStorage?: StorageOptions;
+  storageOptions?: { indexedDBName: string; indexedDBObjectStoreName: string }; // TODO: generalize it to localStorage as well
+}
+
+const cacheInstance = Cache.getInstance();
+
 export const getEmbedding = async (
   text: string,
   precision: number = 7,
-  options = { pooling: "mean", normalize: false },
-  model = "Xenova/gte-small"
+  options = { pooling: 'mean', normalize: false },
```

Comment:
update to the latest model initialization
`let transformersModule: any;
let pipeline: any;
let pipe: any;
let currentModel: string;

export const initializeModel = async (model: string = "Xenova/gte-small"): Promise<void> => {
  if (model !== currentModel) {
    transformersModule = await import("@xenova/transformers");
    pipeline = transformersModule.pipeline;
    pipe = await pipeline("feature-extraction", model);
    currentModel = model;
  }
}

export const getEmbedding = async (
  text: string,
  precision: number = 7,
  options = { pooling: "mean", normalize: false },
  model: string = "Xenova/gte-small"
): Promise<number[]> => {
  if (model !== currentModel) {
    await initializeModel(model);
  }
  const c
`

Reply:
fixed now, we don't need to map pipeline and pipe, transformersModule is enough on global namespace